assembly/src/assembler/instruction/crypto_ops.rs

```diff
@@ -145,27 +145,41 @@ pub(super) fn mtree_set(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, Ass
     // stack: [d, i, R_old, V_new, ...]
 
     // stack: [V_old, R_new, ...] (29 cycles)
-    update_mtree(span, true)
+    update_mtree(span)
 }
 
-/// Appends the MRUPDATE op with a parameter of "true" and stack manipulations to the span block as
-/// required to copy a Merkle tree with root R and update the node in the copied tree at depth d
-/// and index i to value V. The stack is expected to be arranged as follows (from the top):
-/// - depth of the node, 1 element; this is expected to be the depth of the Merkle tree
-/// - index of the node, 1 element
-/// - current root of the tree, 4 elements
-/// - new value of the node, 4 element
+/// Injects [AdviceInjector::MerkleMerge] into the advice provider. The stack is expected to be
+/// arranged as follows (from the top):
+/// - root of the right tree, 4 elements
+/// - root of the left tree, 4 elements
 ///
-/// After the operations are executed, the stack will be arranged as follows:
-/// - old value of the node, 4 elements
-/// - new root of the tree after the update, 4 elements
+/// The operation will merge the Merkle trees with the provided roots, producing a new merged root
+/// with incremented depth. After the operations are executed, the stack will be arranged as
+/// follows:
+/// - merged root, 4 elements
 ///
-/// This operation takes 29 VM cycles.
-pub(super) fn mtree_cwm(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError> {
-    // stack: [d, i, R_old, V_new, ...]
+/// This operation will fail if any of the input roots doesn't exist as Merkle tree in the advice
```

Comment:
nit: I'd say "if either of the input roots" instead of "if any of the input roots".

Reply:
Fixed